### PR TITLE
C++ basic chat client

### LINKED ISSUE #61: [Feature]: Port Rust chat client to C++
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Rust chat client and server were added in #51.
The chat client should be ported to C++.
Existing server can be used to test against.

This project is part of **Semester of Code**:
https://github.com/jhu-ospo-courses/JHU-EN.601.270/tree/main/Fall_2022#enarx-projects

### Acceptance Criteria

_No response_

### Suggestions for a technical implementation

_No response_

### PR DESCRIPTION
**Pull request by kieranjazy**
Added a C++ chat client demo

Signed-off-by: Kieran <kieranjayes@hotmail.com>

Closes #61

### CODE REVIEW

File: demos/chat-client/c++/main.cpp [hunk 1]
```diff
@@ -0,0 +1,26 @@
+#include <iostream>
+#include <cstdlib>
+#include <unistd.h>
+#include <string>
+#include <cstring>
+
+int main(int argc, char* argv[]) {
+    int envVarCount = atoi(std::getenv("FD_COUNT"));
+    if (envVarCount != 4) {
+        std::cout << "4 environment variables must be passed via Enarx.toml" << std::endl;
```
Comment (rvolosatovs) [inline demos/chat-client/c++/main.cpp:10]:
```suggestion
        std::cout << "expected exactly 4 pre-opened file descriptors" << std::endl;
```

**Review by rvolosatovs — APPROVED** [review]
Small comment

**Comment by rvolosatovs** [comment]
@kieranjazy could you please squash the commit?

**Comment by kieranjazy** [comment]
Squashed to single commit

**Review by rvolosatovs — APPROVED** [review]
Thanks!

**Pull request merged**